### ISSUE TITLE
Possible VRAM memory leak on macOS

### ISSUE THREAD

**Issue by Piipperi** [jellysquid3/Sodium#1269]
The game starts out smoothly with my VRAM in pretty normal 25 - 30% usage, but slowly as I play it creeps up to until it's full, which causes lag and can even cause a full system freeze.

I've only been able to replicate this issue on macOS. The issue does not occur on Windows on the same hardware.

Minecraft 1.18.2 w/Fabric
macOS 12.3.1
Intel Core i7-4960HQ
16 GB DDR3 @ 1600 MHz
AMD Radeon RX 5700 XT 8 GB

Let me know if you need further diagnosis on the issue

**Comment by jellysquid3** [comment]
Are you using a Hackintosh? I'm not aware of any macOS-based system with an RX 5700 XT.

If that's the case, this is almost certainly a graphics driver bug, and I'm doubtful we will be able to fix it.

**Comment by Piipperi** [comment]
> Are you using a Hackintosh? I'm not aware of any macOS-based system with an RX 5700 XT.
> 
> 
> 
> If that's the case, this is almost certainly a graphics driver bug, and I'm doubtful we will be able to fix it.

I'm using a Hackintosh but I am not using any custom graphics optimizations/drivers, macOS offers drivers for RX 5700 XT out of the box.

**Comment by douira** [comment]
Oh now that I see this issue, I've also experienced this before. It made everything freeze and die. The only thing I found could somewhat alleviate the problems is quickly using a keyboard shortcut to shut down the computer before a forced reset is necessary. While sodium seems to be the only thing causing this, it seems near impossible to debug sodium and the graphics driver. If somebody knowledgeable in this comes along maybe they could find something.
